test/integration/user_muting_test.rb

```diff
@@ -0,0 +1,31 @@
+require "test_helper"
+
+class UserMutingTest < ActionDispatch::IntegrationTest
+  def test_muted_messages_are_moved_into_separate_inbox
+    user = create(:user)
+    other_user = create(:user)
+    user.mutes.create(:subject => other_user)
+
+    session_for(user)
+    get inbox_messages_path
+    assert_select "a[href='#{muted_messages_path}']", false
+
+    session_for(other_user)
```

Comment:
I think just `create(:message, ....)` rather than switching sessions here

Reply:
I've transformed the integration test into a system spec and split up the test cases so that no session switch/or sign_in/sign_out is necessary anymore to verify the behaviour.